org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/editor/WTableDirEditor.java

```diff
@@ -655,6 +660,25 @@ public Lookup getLookup()
     	return lookup;
     }
     
+    protected void actionDrill() {
+    	if(getGridField() == null && getGridField().getGridTab() == null)
+    		return;
+    	
+    	String columnName = getGridField().getColumnName();;
+    	String tableName = "";
+    	
+    	if(columnName.endsWith("_ID"))
+    		tableName = columnName.substring(0, columnName.length()-3);
```

Comment:
if getGridField().getGridTab() is not null, you can get table name from getGridField().getGridTab(). Safer than the substring approach.

Reply:
That would not give me the correct table name, since I need the target table (e.g. when drilling from sales order window, sales rep field, then I need AD_User, not C_Order). 

In the latest commit I used here similar approach to WSearchEditor by adding the setTableAndKeyColumn() method